### PR TITLE
[risk=low][no ticket] Update BillingProjectBufferStatus for multi-tier

### PR DESCRIPTION
**Pull request by jmthibault79**
Description:

This endpoint is not used by the UI or any service, only as debugging info.  So it's safe to update in-place.
 

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/src/main/resources/workbench-api.yaml [hunk 1]
```diff
@@ -5981,10 +5981,21 @@ definitions:
...
+  AvailableBufferPerTier:
+    type: object
+    properties:
+      accessTierShortName:
+        description: 'The shortName of an access tier'
+        type: string
       bufferSize:
-        description: 'Current size of the billing buffer, i.e. the number of projects
-          ready to be claimed.'
+        description: 'Count of the AVAILABLE billing buffer projects in this access tier'
```
Comment (jmthibault79) [inline api/src/main/resources/workbench-api.yaml:5998]: previous version also did this

File: api/src/test/java/org/pmiops/workbench/billing/BillingProjectBufferServiceTest.java [hunk 2]
```diff
@@ -132,31 +135,23 @@ public WorkbenchConfig workbenchConfig() {
 
   @Autowired private BillingProjectBufferService billingProjectBufferService;
 
-  private final String REGISTERED_TIER_NAME = "registered";
-  private final String REGISTERED_TIER_SERVICE_PERIMETER = "registered/tier/perimeter";
   private final int REGISTERED_TIER_BUFFER_CAPACITY = 5;
 
-  private DbAccessTier registeredTier =
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/billing/BillingProjectBufferServiceTest.java:139]: guess I missed this on a previous refactoring sweep.

File: api/src/main/java/org/pmiops/workbench/billing/BillingProjectBufferService.java [hunk 3]
```diff
@@ -326,9 +326,16 @@ private int getBufferMaxCapacity(String accessTierShortName) {
...
-    return new BillingProjectBufferStatus().bufferSize(bufferSize);
+    return new BillingProjectBufferStatus()
+        .availablePerTier(
+            accessTierService.getAllTiers().stream()
+                .map(
+                    tier ->
+                        new AvailableBufferPerTier()
+                            .accessTierShortName(tier.getShortName())
+                            .bufferSize(
+                                billingProjectBufferEntryDao.countByStatusAndAccessTier(
```
Comment (yonghaoy) [inline api/src/main/java/org/pmiops/workbench/billing/BillingProjectBufferService.java:337]: Can we make this in one query, i.e. group by status and tier?

Comment (jmthibault79) [reply api/src/main/java/org/pmiops/workbench/billing/BillingProjectBufferService.java]: good idea.  done.

**Comment by jmthibault79** [comment]
Output looks like
```
{
  "availablePerTier": [
    {
      "accessTierShortName": "registered",
      "availableProjects": 10
    },
    {
      "accessTierShortName": "controlled",
      "availableProjects": 10
    }
  ]
}
```

**Review by yonghaoy — APPROVED** [review]

**Pull request merged**